Update user guide (#124)
AY2223S1-CS2103T-T13-1/tp: docs/UserGuide.md

---

## Quick start
1. Ensure you have Java `11` or above installed in your Computer.
2. Download the latest TAA.jar from [here](https://github.com/AY2223S1-CS2103T-T13-1/tp/releases/tag/v1.3.trial).
3. Copy the file to the folder you want to use as the home folder for your TAA.
4. Double-click the file to start the app. The GUI appear in your app should be similar as the one shown below:
   <img src="images/TAA.png" width="500">
5. Type the command in the command box and press Enter or click the Send button to execute. Some example commands you can try:

> * `student add` `n/John p.96123456 e/john@example.com g/T03`: Adds a student named John to the TAA.

Reviewer: ```suggestion
* `student add` `n/John p/96123456 e/john@example.com g/T03`: Adds a student named John to the TAA.
```

> * `student edit` `1 g/T01`: Edits the student John to change its tutorial group from T03 to T01.

Reviewer: ```suggestion
* `student edit` `1 g/T01`: Edits the student John to change his tutorial group from T03 to T01.
```

* `student delete` `1`: Removes the student John from TAA.
6. Refer to the Features below for details of each command.
---

## Features

### Add students : 

- Command `student add n/studentName p/phoneNumber e/email g/tutorialGroup(optional) t/tags(optional)`
- Adds a student with the given phone number, email, tutorial group, and tags to the list of students

Notes:
- `<studentName>` should be alphanumeric and should not be blank
- `<phoneNumber>` should only contain numbers, and it should be at least 3 digits long
- `<email>` should contain in the format: local-part@domain
  - local-part should only contain alphanumerics and these special characters, excluding the parentheses, (+_.-) 
  - local-part should not start or end with any special characters
  - domain should be at least 2 characters long
- `<tutorialGroup>` should follow the format Txx, where x is a numeric value, and it should not be blank
- `<tags>` should be alphanumeric with no white space

### Remove students : 

- Command `student delete <index>`
- Removes the student from the list of students with the given index

Notes:

> - `index` must be a positive integer and valid

Reviewer: ```suggestion
- `index` must be a positive integer and must correspond to an existing student
```

### Edit students : 

- Command `student edit <index> n/<studentName>(optional) p/<studentPhone>(optional) e/<studentEmail>(optional) g/<tutorialGroup>(optional) t/<tag>(optional)`
- Edits the student by its given index with at least 1 variable specified to change.

Notes:

> - `index` must be a positive integer and valid

Reviewer: ```suggestion
- `index` must be a positive integer and must correspond to an existing student
```

### List students : 

- Command `student list`
- Abbreviate with `students`
- Shows a list of all students

### Add new task : 

- Command `task add tn/<taskName> i/<taskDescription> d/<taskDeadline> s/<student(s)>(optional)`
- Adds a task to the list of tasks
- The task is initially assigned to no students if no student variable is entered

Notes:
- `taskName` should only contain alphanumeric characters and spaces, and it should not be blank
- `taskDescription` should not be blank
- `taskDeadline` should be in the format of DD/MM/YYYY with its days and months within range

> - `student(s)` should exist

Reviewer: ```suggestion
- `student(s)`, referenced by their name, should exist
```

### Remove task : 

- Command `task delete <index>`

> - Removes the named task from the list of tasks

Reviewer: ```suggestion
- Removes the indexed task from the list of tasks
```
